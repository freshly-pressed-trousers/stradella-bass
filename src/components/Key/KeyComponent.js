import React from "react";
import styles from "./KeyComponent.module.css";
import classNames from "classnames";

function isBlackNote(midiNote) {
	return [1, 3, 6, 8, 10].indexOf(midiNote % 12) >= 0;
}

export default function KeyComponent(props) {
	const { midiNote, isOn, triggerNote, releaseNote } = props;

	return (
		<div
			className={classNames(styles.wrapper, {
				[styles.blackNote]: isBlackNote(midiNote),
				[styles.on]: isOn,
			})}
			onPointerDown={() => triggerNote(midiNote)}
			onPointerUp={() => releaseNote(midiNote)}
			onPointerLeave={() => releaseNote(midiNote)}
		/>
	);
}
