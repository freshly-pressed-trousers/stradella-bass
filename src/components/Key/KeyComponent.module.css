.wrapper {
	font-size: 2rem;
	height: 5em;
	border: 1px solid black;
	width: 1em;
	box-sizing: border-box;
	transition: background-color 0.2s, transform 0.2s;
	cursor: pointer;
	background-color: white;
	-webkit-tap-highlight-color: transparent;
	user-select: none;
}

.blackNote {
	background-color: black;
	margin-left: -0.3em;
	margin-right: -0.3em;
	height: 3.5em;
	width: 0.6em;
	border-bottom-left-radius: 2px;
	border-bottom-right-radius: 2px;
	transform: translateZ(3px);
}

.wrapper:hover {
	background-color: red;
}

.on,
.on:hover {
	background-color: green;
	transform: translateZ(-2px);
}
.blackNote.on {
	transform: translateZ(1px);
}
