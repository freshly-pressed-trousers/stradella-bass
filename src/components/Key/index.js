import KeyComponent from "./KeyComponent";
import NoteContainer from "containers/Note";

const ConnectedKeyComponent = NoteContainer(KeyComponent);

export default ConnectedKeyComponent;
