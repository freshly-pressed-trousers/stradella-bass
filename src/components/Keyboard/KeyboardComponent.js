import React from "react";
import Key from "components/Key";
import styles from "./KeyboardComponent.module.css";

const BASE_NOTE = 45;
const KEYS = 33;

export default function ButtonRowComponent(props) {
	return (
		<div className={styles.wrapper}>
			{new Array(KEYS).fill(true).map((v, index) => (
				<Key key={index} midiNote={BASE_NOTE + index} />
			))}
		</div>
	);
}
