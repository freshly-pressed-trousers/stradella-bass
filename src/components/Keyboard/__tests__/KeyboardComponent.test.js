import React from "react";
import ReactDOM from "react-dom";
import KeyboardComponent from "../KeyboardComponent";
import { shallow } from "enzyme";
import Key from "components/Key";

describe("Keyboard component", () => {
	it("matches snapshot", () => {
		expect(shallow(<KeyboardComponent />)).toMatchSnapshot();
	});

	it("has 33 keys", () => {
		const component = shallow(<KeyboardComponent />);

		expect(component.find(Key).length).toEqual(33);
	});
});
