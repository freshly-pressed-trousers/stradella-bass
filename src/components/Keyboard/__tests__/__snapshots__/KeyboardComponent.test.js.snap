// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Keyboard component matches snapshot 1`] = `
<div
  className="wrapper"
>
  <Connect(KeyComponent)
    key="0"
    midiNote={45}
  />
  <Connect(KeyComponent)
    key="1"
    midiNote={46}
  />
  <Connect(KeyComponent)
    key="2"
    midiNote={47}
  />
  <Connect(KeyComponent)
    key="3"
    midiNote={48}
  />
  <Connect(KeyComponent)
    key="4"
    midiNote={49}
  />
  <Connect(KeyComponent)
    key="5"
    midiNote={50}
  />
  <Connect(KeyComponent)
    key="6"
    midiNote={51}
  />
  <Connect(KeyComponent)
    key="7"
    midiNote={52}
  />
  <Connect(KeyComponent)
    key="8"
    midiNote={53}
  />
  <Connect(KeyComponent)
    key="9"
    midiNote={54}
  />
  <Connect(KeyComponent)
    key="10"
    midiNote={55}
  />
  <Connect(KeyComponent)
    key="11"
    midiNote={56}
  />
  <Connect(KeyComponent)
    key="12"
    midiNote={57}
  />
  <Connect(KeyComponent)
    key="13"
    midiNote={58}
  />
  <Connect(KeyComponent)
    key="14"
    midiNote={59}
  />
  <Connect(KeyComponent)
    key="15"
    midiNote={60}
  />
  <Connect(KeyComponent)
    key="16"
    midiNote={61}
  />
  <Connect(KeyComponent)
    key="17"
    midiNote={62}
  />
  <Connect(KeyComponent)
    key="18"
    midiNote={63}
  />
  <Connect(KeyComponent)
    key="19"
    midiNote={64}
  />
  <Connect(KeyComponent)
    key="20"
    midiNote={65}
  />
  <Connect(KeyComponent)
    key="21"
    midiNote={66}
  />
  <Connect(KeyComponent)
    key="22"
    midiNote={67}
  />
  <Connect(KeyComponent)
    key="23"
    midiNote={68}
  />
  <Connect(KeyComponent)
    key="24"
    midiNote={69}
  />
  <Connect(KeyComponent)
    key="25"
    midiNote={70}
  />
  <Connect(KeyComponent)
    key="26"
    midiNote={71}
  />
  <Connect(KeyComponent)
    key="27"
    midiNote={72}
  />
  <Connect(KeyComponent)
    key="28"
    midiNote={73}
  />
  <Connect(KeyComponent)
    key="29"
    midiNote={74}
  />
  <Connect(KeyComponent)
    key="30"
    midiNote={75}
  />
  <Connect(KeyComponent)
    key="31"
    midiNote={76}
  />
  <Connect(KeyComponent)
    key="32"
    midiNote={77}
  />
</div>
`;
